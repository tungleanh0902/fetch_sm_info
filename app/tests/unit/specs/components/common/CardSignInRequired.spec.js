import { shallowMount } from "@vue/test-utils"
import CardSignInRequired from "common/CardSignInRequired"

describe(`CardSignInRequired`, () => {
  let wrapper, $store

  beforeEach(() => {
    const state = {
      session: {
        insecureMode: false,
        browserWithLedgerSupport: null,
      },
      extension: {
        enabled: true,
      },
      keystore: {
        accounts: [],
      },
    }
    $store = {
      state,
      commit: jest.fn(),
      dispatch: jest.fn(() => Promise.resolve()),
      getters: {
        currentNetwork: { id: `polka-mon`, network_type: `polkadot` },
      },
    }
    wrapper = shallowMount(CardSignInRequired, {
      mocks: {
        $store,
      },
      stubs: [`router-link`],
    })
  })

  it(`shows a sign in required card`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
