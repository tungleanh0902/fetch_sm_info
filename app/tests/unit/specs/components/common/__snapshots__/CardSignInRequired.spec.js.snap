// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSignInRequired shows a sign in required card 1`] = `
<sessionframe-stub
  icon="account_box"
>
  <div
    class="card-sign-in"
  >
    <h2>
      Welcome to Lunie ðŸ‘‹
    </h2>
     
    <h3>
      How would you like to get started?
    </h3>
     
    <div
      class="session-list"
    >
      <!---->
       
      <!---->
       
      <lisession-stub
        icon="language"
        id="explore-with-address"
        img=""
        route="/explore"
        title="Explore with any address"
      />
       
      <!---->
    </div>
     
    <router-link-stub
      class="footnote"
      to="/create"
    >
      
      Want to create a new address?
    
    </router-link-stub>
  </div>
</sessionframe-stub>
`;
